src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentPresentation/TextDocumentUriPresentationEndpoint.cs

```diff
@@ -81,24 +81,26 @@ protected override IRazorPresentationParams CreateRazorRequestParameters(UriPres
             return null;
         }
 
+        var razorFileUri = request.Uris.Where(
+            x => Path.GetFileName(x.GetAbsoluteOrUNCPath()).EndsWith(".razor", FilePathComparison.Instance)).LastOrDefault();
+
         // We only want to handle requests for a single .razor file, but when there are files nested under a .razor
         // file (for example, Goo.razor.css, Goo.razor.cs etc.) then we'll get all of those files as well, when the user
         // thinks they're just dragging the parent one, so we have to be a little bit clever with the filter here
-        var razorFileUri = request.Uris.Last();
-        var fileName = Path.GetFileName(razorFileUri.GetAbsoluteOrUNCPath());
-        if (!fileName.EndsWith(".razor", FilePathComparison.Instance))
+        if (razorFileUri == null)
         {
-            _logger.LogInformation("Last file in the drop was not a single razor file URI.");
+            _logger.LogInformation("No file in the drop was a razor file URI.");
             return null;
         }
 
+        var fileName = Path.GetFileName(razorFileUri!.GetAbsoluteOrUNCPath());
```

Comment:
Is the exclamation mark here and on line 103 required? Seems surprising

Reply:
Removed it